src/llm/language_model/legacy/servable.cpp

```diff
@@ -98,6 +98,9 @@ absl::Status LegacyServable::readCompleteExecutionResults(std::shared_ptr<GenAiS
         return absl::CancelledError();
     }
     legacyExecutionContext->finished.wait();
+    if(legacyExecutionContext->errorMessage.size() > 0){
+        return absl::InvalidArgumentError(legacyExecutionContext->errorMessage);
```

Comment:
Is it readable? I wonder if we don't share too much internal details here. Maybe we can log error message, but return more generic message to the user.